### PR TITLE
Fix some rizin memory leaks

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/bin/bobj.c [hunk 1]
```diff
@@ -315,21 +315,25 @@ RZ_API int rz_bin_object_set_items(RzBinFile *bf, RzBinObject *o) {
 	if (p->fields) {
 		o->fields = p->fields(bf);
 		if (o->fields) {
-			o->fields->free = rz_bin_field_free;
+			if (!o->fields->free) {
```
Comment (ret2libc) [inline librz/bin/bobj.c]: Where does this happen? I'd say that plugins are expected to pass a RzList that is properly created with the right free callback. If this does not happen, it's the plugin's fault, not of RzBin. WDYT?

**Comment by XVilka** [comment]
@wargio you mentioned in Java PR that `mdmp` test crashes, looks like the cause is somewhere in this PR instead:
```
[XX] db/formats/mdmp : test format definitions
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -nn -Qc pf. bins/mdmp/hello.dmp
-- stdout
@@ -1,21 +0,0 @@
-pf.mdmp_directory [4]E? (mdmp_stream_type)StreamType (mdmp_location_descriptor)Location
-pf.mdmp_exception [4]E[4]Eqqdd[15]q (mdmp_exception_code)ExceptionCode (mdmp_exception_flags)ExceptionFlags ExceptionRecord ExceptionAddress NumberParameters __UnusedAlignment ExceptionInformation
-pf.mdmp_exception_stream dd?? ThreadId __Alignment (mdmp_exception)ExceptionRecord (mdmp_location_descriptor)ThreadContext
-pf.mdmp_header [4]zddddt[8]B Signature Version NumberOfStreams StreamDirectoryRVA CheckSum TimeDateStamp (mdmp_type)Flags
-pf.mdmp_location_descriptor dd DataSize RVA
-pf.mdmp_location_descriptor64 qq DataSize RVA
-pf.mdmp_memory64_list qq[83]? NumberOfMemoryRanges BaseRva (mdmp_memory_descriptor64)MemoryRanges
-pf.mdmp_memory_descriptor q? StartOfMemoryRange (mdmp_location_descriptor)Memory
-pf.mdmp_memory_descriptor64 qq StartOfMemoryRange DataSize
-pf.mdmp_memory_info qq[4]Edq[4]E[4]E[4]Ed BaseAddress AllocationBase (mdmp_page_protect)AllocationProtect __Alignment1 RegionSize (mdmp_mem_state)State (mdmp_page_protect)Protect (mdmp_mem_type)Type __Alignment2
-pf.mdmp_memory_info_list ddq[127]? SizeOfHeader SizeOfEntry NumberOfEntries (mdmp_memory_info)MemoryInfo
-pf.mdmp_misc_info d[4]Bdtttddddd SizeOfInfo (mdmp_misc1_flags)Flags1 ProcessId ProcessCreateTime ProcessUserTime ProcessKernelTime ProcessorMaxMhz ProcessorCurrentMhz ProcessorMhzLimit ProcessorMaxIdleState ProcessorCurrentIdleState
-pf.mdmp_module qddtd???qq BaseOfImage SizeOfImage CheckSum TimeDateStamp ModuleNameRVA (mdmp_vs_fixedfileinfo)VersionInfo (mdmp_location_descriptor)CvRecord (mdmp_location_descriptor)MiscRecord Reserved0 Reserved1
-pf.mdmp_module_list d[10]? NumberOfModule (mdmp_module)Modules
-pf.mdmp_string dZ Length Buffer
-pf.mdmp_system_info [2]EwwbBddd[4]Ed[2]Ew[2]q (mdmp_processor_architecture)ProcessorArchitecture ProcessorLevel ProcessorRevision NumberOfProcessors (mdmp_product_type)ProductType MajorVersion MinorVersion BuildNumber (mdmp_platform_id)PlatformId CsdVersionRva (mdmp_suite_mask)SuiteMask Reserved2 ProcessorFeatures
-pf.mdmp_thread ddddq?? ThreadId SuspendCount PriorityClass Priority Teb (mdmp_memory_descriptor)Stack (mdmp_location_descriptor)ThreadContext
-pf.mdmp_thread_list d[1]? NumberOfThreads (mdmp_thread)Threads
-pf.mdmp_token_info ddq TokenSize TokenId TokenHandle
-pf.mdmp_token_info_list dddd TokenListSize TokenListEntries ListHeaderSize ElementHeaderSize
-pf.mdmp_vs_fixedfileinfo ddddddddddddd dwSignature dwStrucVersion dwFileVersionMs dwFileVersionLs dwProductVersionMs dwProductVersionLs dwFileFlagsMask dwFileFlags dwFileOs dwFileType dwFileSubtype dwFileDateMs dwFileDateLs


-- stderr
[WARN] Invalid or unsupported enumeration encountered 21
[WARN] Invalid or unsupported enumeration encountered 22
Error parsing resource directory
[INFO] Parsing data sections for large dumps can take time, please be patient (but if strings ain't your thing try with -z)!
```
https://builds.sr.ht/~xvilka/job/461879#task-test-0

File: librz/bin/bobj.c (hunk 1 above)
Comment (wargio) [reply librz/bin/bobj.c]: i prefer to have this fail fallback or do you want to have `rz_warn_if_fail()`?

Comment (ret2libc) [reply librz/bin/bobj.c]: I prefer `rz_warn_if_fail`. Plugins should really be fixed, we should not silently "fixing them".

**Review by ret2libc — APPROVED** [review]

**Pull request merged**